### PR TITLE
Documentation - Update supported IDEs and general improvements

### LINKED ISSUE #8101: add documentation: what happens when an update gets aborted
### Related area

Update.h

### Hardware specification

any ESP32 board

### Is your feature request related to a problem?

It is unclear from the documentation what happens when an OTA update is aborted. `Update.h` includes some documentation of the `end` method but nothing about the `abort` method and no mention of power failure during update, etc.

Will the ESP32 be able to continue running seamlessly? Or will the flash content be corrupted? How to recover if a new firmware has been written to flash only in half and the ESP32 does not work anymore?

### Describe the solution you'd like

documentation of the answers mentioned in the problem description

### Describe alternatives you've considered

I interrupted power supply during flashing and my ESP32 continued to work seemlessly with the old image. However, this test is probably not very useful because both firmwares were similar...

### Additional context

_No response_

### I have checked existing list of Feature requests and the Contribution Guide

- [X] I confirm I have checked existing list of Feature requests and Contribution Guide.

### PR DESCRIPTION
**Pull request by lucasssvaz**
## Description of Change

Update supported IDEs and add link to more information about OTA updates.

## Related links

Closes #8101

### CODE REVIEW

**Comment by PBrunot** [comment]
Why are you removing PlatformIO documentation ? It's a great IDE for this project.

**Comment by lucasssvaz** [comment]
cc @VojtechBartoska

**Comment by VojtechBartoska** [comment]
> Why are you removing PlatformIO documentation ? It's a great IDE for this project.

Hello @PBrunot,

we have no doubts about PlatformIO being a great IDE but it's not officially supported by Espressif Systems right now. We are not able to manage and make sure that PlatformIO is supporting our latest version of Arduino ESP32 Core so due to that, we are deleting it from our Documentation to not confuse our Users. For using Arduino ESP32 Core in PlatformIO IDE, please refer to PlatformIO documentation.

**Comment by torntrousers** [comment]
How about instead keep the info but just add a comment saying its independent and not supported by Espressif?

**Comment by VojtechBartoska** [comment]
> How about instead keep the info but just add a comment saying its independent and not supported by Espressif?

Actually, that's not bad approach.

@lucasssvaz Please revert those changes and add please to the https://github.com/espressif/arduino-esp32/blob/master/docs/source/installing.rst#installing-using-platformio Note block with the sentence above PlatformIO is not officially supported.

 you an inspire from https://github.com/espressif/arduino-esp32/blob/master/docs/source/installing.rst#installing-using-platformio

File: docs/source/installing.rst [hunk 1]
```diff
@@ -73,6 +73,8 @@ Installing using PlatformIO
 
 PlatformIO is a professional collaborative platform for embedded development. It has out-of-the-box support for ESP32 SoCs and allows working with Arduino ESP32 as well as ESP-IDF from Espressif without changing your development environment. PlatformIO includes lots of instruments for the most common development tasks such as debugging, unit testing, and static code analysis.
 
+.. warning:: PlatformIO is not officially supported by Espressif. Please report any issues with PlatformIO directly to its developers `the official PlatformIO repositories <https://github.com/platformio>`_.
```
Comment (pedrominatel) [inline docs/source/installing.rst:76]: Please report any issues with PlatformIO directly to its developers on the official `PlatformIO repositories <https://github.com/platformio>`_.

**Review by pedrominatel — COMMENTED** [review]
Just one minor change.

File: docs/source/installing.rst (hunk 1 above)
Comment (lucasssvaz) [reply docs/source/installing.rst:76]: Oops, missed that. Thanks!

**Review by pedrominatel — APPROVED** [review]
LGTM

**Review by VojtechBartoska — APPROVED** [review]
Thanks for update @lucasssvaz

**Comment by ivankravets** [comment]
Oh, we were not aware that Espressif was considering removing PlatformIO from its official documentation. It's surprising given our long collaboration and the significant value that PlatformIO has added to Espressif's business.

In response, we've created a separate issue (https://github.com/platformio/platform-espressif32/issues/1225) to inform the community that version 2.x is the latest supported version. We hope to maintain the support and use of PlatformIO, as many Espressif users prefer it over Arduino IDE.

**Comment by VojtechBartoska** [comment]
Hi @ivankravets, PlatformIO is not removed from our documentation. We just added a note that the support of our core is maintained by PlatformIO and not by us to to make it clear for Users that it is a separate project and we are not able to handle the compatibility between our Core and PIO.

So we did not change anything, just called the current state properly. We will be glad if PIO still keeps support of our latest version.

We are open to discuss this more and make it clear :)

**Comment by thijstriemstra** [comment]
you're doing a really bad job handling this @VojtechBartoska. Creating confusion, reverting things. not sure if you work at espressif but this is a bad look for all of your employees.

**Comment by VojtechBartoska** [comment]
Hello @thijstriemstra,

first of all, yes I do work in Espressif and I am Project Manager of Arduino Core Project. I hope not all contributors of this repository thinks I am doing a bad job, I can say to you I am trying to do my best. 

I am sorry that this Pull Request (change) created confusion to you, it was not meant to do this but exact opposite. 

In general in this project we are trying to be as transparent as possible and we are really focused on user experience and trying to make this framework user friendly.

I will explain to you again, even though it is already explained in comments above.

This project is own and managed by Espressif and has wide base of different User which use different IDEs. 

PlatformIO is one of them, it is separate project providing IDE which we do not own, we are not able to keep it compatible. We have PlatformIO IDE and Arduino.cc IDE mentioned in our documentation page of this project because we want to mention most common (from our limited point of view) IDEs to have complete getting started docs for end user. If any IDE owner/maintainer contribute to the IDEs page in our Docs we will be more than happy.

Regarding this particular PR, Users are still coming to our repository mentioning PlatformIO issues we are not able to solve, due to that we initially though we will delete PIO documentation from our docs because the content was outdated and was bringing even more confusion and we are not able to keep this section up-to-date as we used different IDEs. After reconsidering this and also good feedback from @torntrousers, we decided to keep this PIO outdated docs there and just add a note that PIO is not developed by us. You are right that this note can be written in different way so we will change it. I hope it is now clear.

To summarize this, we will update PIO note in our docs to not bring a confusion, we are open to your feedback.

**Comment by Jason2866** [comment]
At least communication with the Platformio CEO before doing mandatory changes in documentation about support would have been a good idea. I hope for the many Platformio users there will be now a communication to clarify everything and mention Platformio prominent in the documentation as great IDE choice.
Bigger projects simply wouldn't be possible without Platformio. ArduinoIDE is to limited.

**Comment by VojtechBartoska** [comment]
@Jason2866 Yea, we should ping @ivankravets in this PR before merging, we just did not consider adding this note just in our docs page as that major change. Support of 2.X Arduino Core version is still there mentioned by PIO Team. 

In general, there was no intention to hide anything or even upset someone, as I mentioned we just want to clarify we are not responsible and also able to help with PIO related issues.

And again, this project is open-source, anyone can contribute and we are not alway able to ping/let know all related projects maintainers. We will try to improve this, thanks for feedback. 👍

**Comment by thijstriemstra** [comment]
> In general, there was no intention to hide anything or even upset someone

Same here, sorry for sounding harsh, and thanks for rectifying things.

**Comment by timonsku** [comment]
In my own experience it can be a very significant support burden to deal with PIO specific issues.
PIO is a useful tool for many but I think its absolutely a reasonable step to say you can't provide support from your side as PIO is not in your control to fix these issue that people encounter. Its more beneficial for everyone if those problems are directed to the PIO issue tracker directly imo.
If people can't replicate their issue in the Arduino IDE then the issue is elsewhere, making sure of that saves a lot of time in debugging root causes.

**Comment by peteDDD** [comment]
No doubt the Espressif folks were biased toward the (toy) Arduino IDE by the incorporation of ESP32 chips in new Arduino boards.  Money talks...

Espressif is apparently not considering the quality and productivity of the IDEs.  Real developers use real, robust IDEs like Visual Code and Visual Code Studio.  Arduino IDE is a toy in comparison.  The value-added to VSC by the PlatformIO team has been immense and spans many frameworks and platforms.   This allows developers who must support multiple platforms to have one common toolset across those platforms which translates to real productivity for the developer.   

I've been programming microprocessors since the 1970's with the Intel 8080 and studied under the program manager of the Motorola (later MicroChip) 6800 family back then.   I've seen many generations of development tools.  The ESP-IDF toolset has a user interface about as primitive as the tools we used back in the 1970s.   VSC with PlatformIO makes programming and testing ESP32-family-based projects a pleasure in comparison.  Meanwhile the Arduino IDE is for "makers" playing around, not for real product development, testing, and deployment.

**Comment by Jason2866** [comment]
@peteDDD There is nothing to add.

**Comment by rftafas** [comment]
Hi all,
This has already far deviated from the purpose of a PR. In order to keep this repo for its intended objective (sharing code), as well as the PRs (discussion about the PR itself) we will lock this and we suggest moving the discussion to the [proper page](https://github.com/espressif/arduino-esp32/discussions) or another discussion forum.
